tmail-backend/mailbox/team-mailboxes/src/test/scala/com/linagora/tmail/team/TeamMailboxRepositoryContract.scala

```diff
@@ -0,0 +1,280 @@
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, BOB)).block()
+    SMono.fromPublisher(testee.removeMember(TEAM_MAILBOX, BOB)).block()
+
+    assertThat(SFlux.fromPublisher(testee.listMembers(TEAM_MAILBOX)).collectSeq().block().asJava)
+      .doesNotContain(BOB)
+  }
+
+  @Test
+  def removeMemberShouldNotRevokeUnAssignMember(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, BOB)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, ANDRE)).block()
+    SMono.fromPublisher(testee.removeMember(TEAM_MAILBOX, BOB)).block()
+
+    assertThat(SFlux.fromPublisher(testee.listMembers(TEAM_MAILBOX)).collectSeq().block().asJava)
+      .contains(ANDRE)
+  }
+
+  @Test
+  def listMemberShouldReturnEmptyWhenDefault(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    assertThat(SFlux.fromPublisher(testee.listMembers(TEAM_MAILBOX)).collectSeq().block().asJava)
+      .isEmpty()
+  }
+
+  @Test
+  def listMemberShouldReturnListMembersHaveRightsWhenSingleMember(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, BOB)).block()
+
+    assertThat(SFlux.fromPublisher(testee.listMembers(TEAM_MAILBOX)).collectSeq().block().asJava)
+      .containsExactlyInAnyOrder(BOB)
+  }
+
+  @Test
+  def listMemberShouldReturnListMembersHaveRightsWhenSeveralMembers(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, BOB)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, ANDRE)).block()
+
+    assertThat(SFlux.fromPublisher(testee.listMembers(TEAM_MAILBOX)).collectSeq().block().asJava)
+      .containsExactlyInAnyOrder(BOB, ANDRE)
+  }
+
+  @Test
+  def listTeamMailboxesByUserShouldReturnEmptyByDefault(): Unit = {
+    assertThat(SFlux.fromPublisher(testee.listTeamMailboxes(BOB)).collectSeq().block().asJava)
+      .isEmpty()
+  }
+
+  @Test
+  def listTeamMailboxesByUserShouldReturnTeamMailboxesWhichAssignUserInThat(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, BOB)).block()
+
+    assertThat(SFlux.fromPublisher(testee.listTeamMailboxes(BOB)).collectSeq().block().asJava)
+      .containsExactlyInAnyOrder(TEAM_MAILBOX)
+  }
+
+  @Test
+  def listTeamMailboxesByUserShouldNotReturnTeamMailboxesWhichAssignUserNotInThat(): Unit = {
+    SMono.fromPublisher(testee.createTeamMailbox(TEAM_MAILBOX)).block()
+    SMono.fromPublisher(testee.addMember(TEAM_MAILBOX, ANDRE)).block()
+
+    assertThat(SFlux.fromPublisher(testee.listTeamMailboxes(BOB)).collectSeq().block().asJava)
+      .isEmpty()
+  }
+
+  @Test
+  def userCanHaveSeveralTeamMailboxes(): Unit = {
```

Comment:
```suggestion
  def userCanBeMemberOfSeveralTeamMailboxes(): Unit = {
```
Remember a team mailbox does not belong to any user :)